filters/auth/oidc.go

```diff
@@ -200,27 +191,45 @@ func (f *tokenOidcFilter) validateAnyClaims(h map[string]interface{}) bool {
 		return true
 	}
 
-	var a []string
-	for k := range h {
-		a = append(a, k)
+	for i := 0; i < len(f.claims)-1; i += 2 {
+		k := f.claims[i]
+		v := f.claims[i+1]
+		if val, ok := h[k]; ok {
+			vs, ok := val.(string)
+			if !ok {
+				continue
+			}
+			if vs == v {
+				return true
+			}
+		} else {
+			continue
+		}
 	}
-
-	log.Debugf("intersect(%v, %v)", f.claims, a)
-	return intersect(f.claims, a)
+	return false
 }
 
 func (f *tokenOidcFilter) validateAllClaims(h map[string]interface{}) bool {
 	if len(f.claims) == 0 {
 		return true
 	}
 
-	var a []string
-	for k := range h {
-		a = append(a, k)
+	for i := 0; i < len(f.claims)-1; i += 2 {
+		k := f.claims[i]
+		v := f.claims[i+1]
```

Comment:
same as in case of all claims: we should not store expected values in the filter parameters, and docs also states it different.

Reply:
fixed